woocommerce/woocommerce-ios
# Coupons: Update Storage with CouponSearchResult

Part of #5768


### Description

To improve the performance for the search functionality of coupons, this PR updates the Storage layer with a new entity `CouponSearchResult` to persist coupons with matching keywords. This also includes a many-to-many relationship between coupons and search results. 

### Testing instructions

Make sure that the app builds and runs properly without crashing with this new update. The coupon list (navigated from Menu tab > Coupons) should work as before.

### Screenshots

N/A

---
